popup.js

```diff
@@ -132,172 +132,186 @@ function checkUrl() {
-  data[AGENCY_ID] = agencyID;
-  data[SUBAGENCY_ID] = subAgencyID;
-  data[ORGANIZATION_ID] = organizationID;
-  data[SUBORG_ID] = suborgID;
-  data[SUBPRIMER_ID] = subprimerID;
-  //data[CRAWLABLE_ID] = crawlableID;
-  data[FTP_ID] = ftpID ? 'Yes' : '';
-  data[VISUALIZATION_ID] = visualizationID ? 'Yes' : '';
-  data[DIFFICULTY_ID] = difficultyID ? 'Yes' : '';
-  data[DATABASE_ID] = databaseID ? 'Yes' : '';
-  data[COMMMENT_ID] = commentID;
-
-  $.get(GOOGLE_FORMS_URL, data)
+  $('#submit').prop('disabled', true);
+
+  $.when(submitToGoogleForms(fields), submitToUntGwda(fields))
     .then(function () {
       $('#submit').prop('disabled', false);
 
       showStatus('success', 'URL submitted. Thanks!');
       hideStatus(3000);
-      // uncomment this line to also add the URL through the official notificaiton tool.
-      // window.open(NOTIFICATION_TOOL_URL + currentURL);
     })
     .catch(function (error) {
       $('#submit').prop('disabled', false);
-      showStatus('failure', 'Could not submit URL (' + (error.statusText || 'Generic error') + ')');
-    })
+      showStatus('failure', 'Could not submit URL');
+    });
 }
 
 /* When the popup loads: Autopopulate the name, event name and email if it has been submitted before,
 * i.e. if localStorage has these fields already saved.
 */
 $(function () {
-  if ( localStorage.name && localStorage.name !== "null" ) {
-    $( '#name' ).val( localStorage.name );
-  }
-  if ( localStorage.email && localStorage.email !== "null" ) {
-    $( '#email' ).val( localStorage.email );
-  }
-  if ( localStorage.eventName && localStorage.eventName !== "null" ) {
-    $( '#eventName' ).val( localStorage.eventName );
-  }
-  if ( localStorage.agency && localStorage.agency !== "null" ) {
-    $( '#agency' ).val( localStorage.agency );
-  }
+  $('#nominatorName').val(localStorage.nominatorName);
+  $('#nominatorEmail').val(localStorage.nominatorEmail);
+  $('#eventName').val(localStorage.eventName);
+  $('#agencyName').val(localStorage.agencyName);
+  $('#agencyId').val(localStorage.agencyId);
+  $('#subAgencyId').val(localStorage.subAgencyId);
+  $('#organizationId').val(localStorage.organizationId);
+  $('#suborgId').val(localStorage.suborgId);
+  $('#subprimerId').val(localStorage.subprimerId);
 
-  if ( localStorage.agencyID && localStorage.agencyID !== "null" ) {
-    $( '#agencyID' ).val( localStorage.agencyID );
-  }
-  if ( localStorage.subAgencyID && localStorage.subAgencyID !== "null" ) {
-    $( '#subAgencyID' ).val( localStorage.subAgencyID );
-  }
-  if ( localStorage.organizationID && localStorage.organizationID !== "null" ) {
-    $( '#organizationID' ).val( localStorage.organizationID );
-  }
-  if ( localStorage.suborgID && localStorage.suborgID !== "null" ) {
-    $( '#suborgID' ).val( localStorage.suborgID );
-  }
-  if ( localStorage.subprimerID && localStorage.subprimerID !== "null" ) {
-    $( '#subprimerID' ).val( localStorage.subprimerID );
-  }
+  checkAgency();
```

Comment:
Fixed bug: Wasn't doing this check after recovering agency values from localStorage.